### PR TITLE
chore(wasmer): add helpers.go file with helper functions

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

- Split out (CGO related) helper functions from `imports.go` to `lib/runtime/wasmer/helpers.go`
- Move pointer size helper functions to `lib/runtime/wasmer/helpers.go`
- Change `toWasmMemorySized` to NOT take a size argument (unneeded)
- Clarify all comments for helper functions
- Update all error wrappings
- Review variable names
  - Use `ptr` instead of `out` for 32 bit pointers
  - Use `pointerSize` instead of `span` for 64 bit pointers size
  - Name return values
  - Other minor renamings such as `res` to `result`, `enc` to `encodedResult`
- Optimizations:
  - `storageAppend`: use slice capacity allocation, `copy` and remove unneeded `append`s
  - `toKillStorageResultEnum`: use `copy` instead of `append`
  - `toWasmMemoryOptional`: remove unneeded variable copy

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

Friday evening fun.
Also aiming to make the migration to wazero easier.

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

File: lib/runtime/wasmer/helpers.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package wasmer
+
+// #include <stdlib.h>
+//
```
Comment (EclesioMeloJunior) [inline lib/runtime/wasmer/helpers.go:7]:
```suggestion
```

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM, just a suggestion: since all the operations relies on the `wasmer.InstanceContext` could we rename the file from `helpers.go` to `context.go` or `instance_context.go`?

File: lib/runtime/wasmer/helpers.go (hunk 1 above)
Comment (qdm12) [reply lib/runtime/wasmer/helpers.go:7]: Actually that empty line is needed for deepsource to compile successfully. Don't ask me why 😄

**Review by jimjbrettj — APPROVED** [review]
lgtm

**Comment by qdm12** [comment]
> LGTM, just a suggestion: since all the operations relies on the `wasmer.InstanceContext` could we rename the file from `helpers.go` to `context.go` or `instance_context.go`?

Good suggestion, although `storageAppend` isn't really about context, and I also sneaked in pointer size conversion functions in there, so `helpers.go` seems appropriate for now.

**Review by timwu20 — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
FYI, I noticed this was in prep of a possible use of wazero, and thanks for thinking of us.

wazero just cut its first beta tag (v1.0.0-beta.1) and also opened a [gophers slack](https://gophers.slack.com/) `#wazero` channel for support, updates and conversation! Note: You may need an [invite](https://invite.slack.golangbridge.org/) to join gophers.

We've also been working harder on our website for things people usually can't find https://wazero.io/languages/

Wish you well and hope to see you around.

**Comment by qdm12** [comment]
Awesome @codefromthecrypt thanks for letting us know! We're probably a few months away from switching to wazero due to other priorities, but it's definitely something we would all like to jump to for sure!

**Pull request merged**